### ISSUE TITLE
curry() vs curry_args() with single argument, former has errors

### ISSUE THREAD

**Issue by carlwiedemann** [matteosister/php-curry#7]
I would consider the following actions to be equivalent:

```php

require './vendor/autoload.php';

use Cypress\Curry as C;

$identity = function ($arg) {
  return $arg;
};

echo "using curry_args()" . PHP_EOL;
$curried_identity = C\curry_args($identity, [TRUE]);
var_dump($curried_identity());

echo "using curry()" . PHP_EOL;
$curried_identity = C\curry($identity, TRUE);
var_dump($curried_identity());
```

But when I run the above script, `curry()` seems to have trouble while `curry_args()` does not.

```
$ php -f index.php
using curry_args()
bool(true)
using curry()
PHP Warning:  Missing argument 1 for {closure}(), called in /Users/c4rl/_munge/currying/index.php on line 17 and defined in /Users/c4rl/_munge/currying/index.php on line 7

Warning: Missing argument 1 for {closure}(), called in /Users/c4rl/_munge/currying/index.php on line 17 and defined in /Users/c4rl/_munge/currying/index.php on line 7
PHP Notice:  Undefined variable: arg in /Users/c4rl/_munge/currying/index.php on line 8

Notice: Undefined variable: arg in /Users/c4rl/_munge/currying/index.php on line 8
NULL
```

This seems like a bug to me. Is it?

**Comment by matteosister** [comment]
It is!
Gonna check this as soon as possible.
Thank you for taking time reporting this...

**Comment by matteosister** [comment]
should be fixed in 0.5.0

could you please check it out?

**Comment by carlwiedemann** [comment]
Fixed! Thanks for the quick response.
